src/LeafInputSignature.cpp

```diff
@@ -249,6 +248,25 @@ struct HeaderValidationError
     const char* included_file;
 };
 
+bool IsGeneratedOrIsLeafInput (BuildQueue* queue, const Frozen::DagDerived* dagDerived, uint32_t hash, const char* filename, RuntimeNode* runtimeNode)
+{
+    const Frozen::DagNode* generatingNode;
+    if (FindDagNodeForFile(&queue->m_Config.m_DagRuntimeData, hash, filename, &generatingNode))
+        return true;
+    if (HashSetLookup(&runtimeNode->m_CurrentLeafInputSignature->m_ExplicitLeafInputs, hash,filename))
+        return true;
+    if (HashSetLookup(&runtimeNode->m_CurrentLeafInputSignature->m_ImplicitLeafInputs, hash,filename))
+        return true;
+    for (auto& dependentNodeThatIsCacheableItself: dagDerived->DependentNodesThatThemselvesAreLeafInputCacheableFor(runtimeNode->m_DagNodeIndex))
```

Comment:
By "child cacheable node" i mean the scenario if a cacheable node A (in the scenario we saw on katana, UnityPlayer.so) has another cacheable node B as an input (libunwind.a). Any leaf input from A should also be considered a leaf input of B. For the leaf input signature calculation, we handle this correctly by including the signature of B in the signature of A. But it turned out that the code verifying that we don't have invalid (ie not known in advance) leaf inputs did not handle this case correctly, as it would not traverse into B. This PR fixes that.